app/src/pages/settings/GenerativeProvidersCard.tsx

```diff
@@ -0,0 +1,256 @@
+import React, { useMemo, useState } from "react";
+import { graphql, useFragment } from "react-relay";
+import {
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  getSortedRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+
+import { Card, Dialog, DialogContainer } from "@arizeai/components";
+
+import {
+  Button,
+  Flex,
+  Icon,
+  Icons,
+  Input,
+  Label,
+  Text,
+  TextField,
+  View,
+} from "@phoenix/components";
+import {
+  GenerativeProviderIcon,
+  GenerativeProviderIconProps,
+} from "@phoenix/components/generative";
+import { tableCSS } from "@phoenix/components/table/styles";
+import { useCredentialsContext } from "@phoenix/contexts/CredentialsContext";
+import { assertUnreachable } from "@phoenix/typeUtils";
+
+import {
+  GenerativeProvidersCard_data$data,
+  GenerativeProvidersCard_data$key,
+} from "./__generated__/GenerativeProvidersCard_data.graphql";
+
+type ProviderKey =
+  GenerativeProvidersCard_data$data["modelProviders"][number]["key"];
+
+function ProviderIcon({ providerKey }: { providerKey: ProviderKey }) {
+  let provider: GenerativeProviderIconProps["provider"];
+  switch (providerKey) {
```

Comment:
Is this necessary? They all map to the same value

Reply:
nope, legacy